python/ray/util/sgd/torch/examples/train_example.py

```diff
@@ -55,20 +54,32 @@ def scheduler_creator(optimizer, config):
 
 def data_creator(config):
     """Returns training dataloader, validation dataloader."""
-    return LinearDataset(2, 5),  LinearDataset(2, 5, size=400)
-
-
-def train_example(num_replicas=1, use_gpu=False):
+    train_dataset = LinearDataset(2, 5, size=config.get("data_size", 1000))
+    val_dataset = LinearDataset(2, 5, size=config.get("val_size", 400))
+    train_loader = torch.utils.data.DataLoader(
```

Comment:
I still think I very much agree with providing a default dataloader thing. Maybe as a util function?

Reply:
I think most likely people will be moving from `torch code` -> `raysgd code`. 

the way I see this being done is:

```python
from torch.utils.data import DataLoader

def data_creator(...):
    train_dataset = LinearDataset(2, 5, size=config.get("data_size", 1000))
    val_dataset = LinearDataset(2, 5, size=config.get("val_size", 400))
    return DataLoader(train_dataset, 32), DataLoader(val_dataset, 32)
```

Reply:
I would like the default to include `config["batch_size"]` or similar in some way, so maybe we should ship a util function simply because there are no reasons not to since it's going to be a one-liner that will require minimal maintenance?If you'd rather keep the API cleaner, it's really not a big deal.

Reply:
Hm, ok. I think your suggestion is reasonable; what would an example API be?

Reply:
Maybe something like this (based on train_example.py)?:
```python
from ray.util.sgd.torch.helpers import default_dataloaders

def data_creator(config):
    """Returns training dataloader, validation dataloader."""
    train_dataset = LinearDataset(2, 5, size=config.get("data_size", 1000))
    val_dataset = LinearDataset(2, 5, size=config.get("val_size", 400))

	# if default_batch_size is unspecified, ValueError if "batch_size" not in config
    return default_dataloaders((train_dataset, val_dataset), config, default_batch_size=32)
```

Note that in you example it doesn't really help:
```python
from ray.util.sgd.torch.helpers import default_dataloaders
 
def data_creator(...):
    train_dataset = LinearDataset(2, 5, size=config.get("data_size", 1000))
    val_dataset = LinearDataset(2, 5, size=config.get("val_size", 400))

	# if default_batch_size is unspecified, ValueError if "batch_size" not in config
    return default_dataloaders((train_dataset, val_dataset), config, default_batch_size=32)
```

Pros are that you won't forget to expose (and use) `batch_size` through the config. Pretty much everyone includes `--batch-size` or equivalent in their CLI (including my upcoming default CLI). This also technically reduces code duplication with having `config.get("batch_size", some_default)` and two calls to `DataLoader`.

Overall I'm suggesting this because I expect that we will end up including many more small snippets like this to ease development. If you think that there won't be any other things that belong in `ray.util.sgd.torch.helpers`, we shouldn't implement `default_dataloaders`.

Reply:
what about:

```python
from ray.util.sgd.torch import as_dataloader, BATCH_SIZE # or TOTAL_BATCH

@as_dataloader(**data_loader_args)  # batch_size can be overriden
def data_creator(...):
    train_dataset = LinearDataset(2, 5, size=config.get("data_size", 1000))
    val_dataset = LinearDataset(2, 5, size=config.get("val_size", 400))
    return train_dataset, val_dataset

trainer = TorchTrainer(
    data_creator=data_creator,
    config={BATCH_SIZE: 128},
    num_workers=4, # or 64
)
trainer.train() 
    
```

Reply:
opened new issue to track this - https://github.com/ray-project/ray/issues/7495